runtime/scripts.vim

```diff
@@ -35,10 +35,12 @@ let s:line1 = getline(1)
 if s:line1 =~# "^#!"
   " A script that starts with "#!".
 
-  " Check for a line like "#!/usr/bin/env VAR=val bash".  Turn it into
-  " "#!/usr/bin/bash" to make matching easier.
+  " Check for a line like "#!/usr/bin/env VAR=val bash" and/or
+  " "#!/usr/bin/env -S bash --norc".  Turn it into "#!/usr/bin/bash"
+  " to make matching easier.
   if s:line1 =~# '^#!\s*\S*\<env\s'
     let s:line1 = substitute(s:line1, '\S\+=\S\+', '', 'g')
+    let s:line1 = substitute(s:line1, '\(-S\|--split-string\)', '', '')
     let s:line1 = substitute(s:line1, '\<env\s\+', '', '')
```

Comment:
Hmm.. this turns `/usr/bin/env bash` into `/usr/bin/bash`, but should rather use just "bash" then maybe directly?
(for correctness `exepath("bash")` should be used, but the path is not really used then later).

Reply:
This line/PR doesn't add that. That's the way it's already done, as per the comment above:

https://github.com/vim/vim/blob/92e25ab2df03756bbb8523cf871db29fe8819c13/runtime/scripts.vim#L38-L39